### ISSUE TITLE
"select scroll" not working properly when you hit edge

### ISSUE THREAD

**Issue by rohrlich** [goki/gide#194]
To be clear :)
While holding the shift key when scrolling you get to a point where your cursor is at the edge of the content and to scroll further you need to move your mouse the opposite direction and sort of ratchet along. The code should detect you are at the edge and keep scrolling as long as you remain at the edge. Textview and the console both exhibit this problem.

**Comment by rcoreilly** [comment]
yeah the problem here is that it is event driven, so if no external events are coming in, there isn't anything driving the "mouse at edge of region" detector.  we would need to setup a timer to generate events in this case, and then turn off that timer at some point.. I guess it could be a pretty generic timer that is used in various situations, and then when those timer events hit the system detects if there is any need for them, and if no longer needed, turns off.. shouldn't be too bad.
